### PR TITLE
Better aliasing for commands

### PR DESCRIPTION
**Pull request by joshuaskelton**
Working on a better way to resolve command/subcommand aliases.

### CODE REVIEW

File: src/taco-cli/cli/commands.json [hunk 1]
```diff
@@ -265,19 +264,14 @@
...
-            ,
-            {
-                "alias": "ls", "command": "list"
+            "aliases": {
+                "plugins": "plugin",
+                "plugin": "plugin",
+                "remove": "remove",
+                "rm": "remove",
+                "list": "list",
+                "ls": "list"
```
Comment (MSLaguana) [inline src/taco-cli/cli/commands.json]: Should we include `"add": "add"` for completeness, even though it doesn't have any 'aliases' as such?

File: src/taco-cli/cli/commands.json [hunk 2]
```diff
@@ -319,15 +313,12 @@
                 "description": "[CommandRemoteListSubcommandDescription]"
             }
             ],
-            "aliases": [
-            {
-                "alias": "rm", "command": "remove"
+            "aliases": {
+                "remove": "remove",
+                "rm": "rm",
```
Comment (MSLaguana) [inline src/taco-cli/cli/commands.json]: should be `"rm": "remove"`

Comment (joshuaskelton) [inline src/taco-cli/cli/commands.json]: Good catch.

File: src/taco-utils/helpCommandBase.ts [hunk 3]
```diff
@@ -257,11 +264,10 @@ module TacoUtility {
          */
         private commandExists(command: string): boolean {
             for (var i in this.commandsFactory.listings) {
-                if (i === command) {
+                if (i === command || (this.commandsFactory.listings[i].aliases && (i === this.commandsFactory.listings[i].aliases[command]))) {
```
Comment (MSLaguana) [inline src/taco-utils/helpCommandBase.ts]: This surprised me; I thought that command-level aliasing (template ~ templates, platform ~ platforms) was done based on the "aliases" entry of commands.json, but this does it by putting both command and subcommand aliases together in each command.

It works how it is at the moment, but I prefer the idea of having a single place where command-level aliases go.

**Comment by MSLaguana** [comment]
Looks good to me.

**Pull request merged**